src/main/java/rx/Single.java

```diff
@@ -1789,4 +1790,36 @@ public void onNext(T t) {
         return zip(this, other, zipFunction);
     }
 
+    /**
+     * Modifies the source {@link Single} so that it invokes an action when it calls {@code onSuccess}.
+     * <p>
+     * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/doOnNext.png" alt="">
+     * <dl>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>{@code doOnSuccess} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     *
+     * @param onSuccess
+     *            the action to invoke when the source {@link Single} calls {@code onSuccess}
+     * @return the source {@link Single} with the side-effecting behavior applied
+     * @see <a href="http://reactivex.io/documentation/operators/do.html">ReactiveX operators documentation: Do</a>
+     */
+    public final Single<T> doOnSuccess(final Action1<? super T> onSuccess) {
```

Comment:
Okay, no problem. I'll mark it as @Experimental.

чт, 8 Окт 2015, 12:44, David Karnok <notifications@github.com>:

> In src/main/java/rx/Single.java
> <https://github.com/ReactiveX/RxJava/pull/3417#discussion_r41494676>:
>
> > @@ -1789,4 +1790,36 @@ public void onNext(T t) {
> >          return zip(this, other, zipFunction);
> >      }
> >
> > +    /**
> > +     * Modifies the source {@link Single} so that it invokes an action when it calls {@code onSuccess}.
> > +     * <p>
> > +     * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/doOnNext.png" alt="">
> > +     * <dl>
> > +     *  <dt><b>Scheduler:</b></dt>
> > +     *  <dd>{@code doOnSuccess} does not operate by default on a particular {@link Scheduler}.</dd>
> > +     * </dl>
> > +     *
> > +     * @param onSuccess
> > +     *            the action to invoke when the source {@link Single} calls {@code onSuccess}
> > +     * @return the source {@link Single} with the side-effecting behavior applied
> > +     * @see <a href="http://reactivex.io/documentation/operators/do.html">ReactiveX operators documentation: Do</a>
> > +     */
> > +    public final Single<T> doOnSuccess(final Action1<? super T> onSuccess) {
>
> Yep, this is a bit of a conflict because #3401
> <https://github.com/ReactiveX/RxJava/pull/3401> gets merged first, this
> would also become beta without any votes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3417/files#r41494676>.
>
-- 

@artem_zin